R/Lithuania.R

```diff
@@ -0,0 +1,409 @@
+        select(
+          date, region_level_1, region_level_2,
+          cases_new, cases_total, deaths_new,
+          tested_new, recovered_total,
+          everything()
+        )
+      # If we have not been asked to return all the OSP-provided data,
+      # just select the core data sought by get_regional_data
+      # (default is FALSE)
+      if (!self$all_osp_fields) {
+        self$data$clean <- self$data$clean %>%
+          select(
+            date, region_level_1, region_level_2,
+            level_1_region_code, level_2_region_code,
+            cases_new, cases_total, deaths_new, tested_new, recovered_total
+          )
+      }
+      if (self$level == "1") {
+        self$clean_level_1()
+      } else if (self$level == "2") {
+        self$clean_level_2()
+      }
+    },
+
+    #' @description Lithuania Specific County Level Data Cleaning
+    #'
+    #' Aggregates data to the level 1 (county) regional level. Data is
+    #' provided by the source at the level 2 (municipality) regional level.
+    #'
+    #' @importFrom dplyr group_by summarise ungroup full_join across if_else
+    #' @importFrom tidyselect vars_select_helpers
+    clean_level_1 = function() {
+
+      self$data$clean <- self$data$clean %>%
+        group_by(.data$date,
+                        .data$region_level_1, .data$level_1_region_code) %>%
+        summarise(
+          across(tidyselect::vars_select_helpers$where(is.numeric),
+                        sum)) %>%
+        mutate(region_level_1
+                      = if_else(is.na(.data$region_level_1),
+                                       "Lietuva", .data$region_level_1
+                      )) %>%
+        ungroup()
+      # Fix percentages, where necessary
+      if (self$all_osp_fields) {
+        # For each of the test percentage fields, recalculate the percentages,
+        # first checking that the total number of checks is not zero.
+        self$data$clean <- self$data$clean %>%
+          mutate(
+            ab_prc_day =
+              if_else(
+                .data$ab_tot_day == 0, 0,
+                .data$ab_pos_day / .data$ab_tot_day
+              ),
+            ag_prc_day =
+              if_else(
+                .data$ag_tot_day == 0, 0,
+                .data$ag_pos_day / .data$ag_tot_day
+              ),
+            pcr_prc_day =
+              if_else(
+                .data$pcr_tot_day == 0, 0,
+                .data$pcr_pos_day / .data$pcr_tot_day
+              ),
+            dgn_prc_day =
+              if_else(
+                .data$dgn_tot_day == 0, 0,
+                .data$dgn_pos_day / .data$dgn_tot_day
+              )
+          )
+      }
+    },
+
+    #' @description Lithuania Specific Municipality Level Data Cleaning
+    #'
+    #' No data cleaning is necessary specific to the municipality level
+    #' since the data is already provided at the municipality level.
+    #'
+    clean_level_2 = function() {
+    },
+
```

Comment:
Remove as not needed by the looks of things

Reply:
Done.